src/main/java/com/eighthlight/fabrial/http/file/FileHttpResponder.java

```diff
@@ -186,4 +188,15 @@ private Response buildPutResponse(Request request, ResponseBuilder builder) {
       return builder.withStatusCode(400).withReason(e.getMessage()).build();
     }
   }
+
+  private Response buildDeleteResponse(Request request, ResponseBuilder builder) {
+    try {
+      fileController.removeFile(request.uri.getPath());
```

Comment:
The method name here `buildDeleteResponse` feels a bit off, since it's performing this `removeFile` effect in addition to building the response (similar w/ `buildPutResponse` above). Maybe `handleDelete` or `performDelete` or something, to reflect it being an action? Situations like this are tough for naming, where you have both a command and a query going on. I guess I tend to assume the query side gets conveyed by the return type / value, and that the action side gets conveyed by the name.

I presume the strict CQRS folks might look for ways to pull this apart into 2 separate methods, but it doesn't seem trivial here due to the exception handling. I do like the way the "router" delegates to a single method here, whatever it's called.

#nonblocker

Reply:
As you mentioned, I think it comes down to the complexity of extricating the response building from the handling of the actual request. Sure, one could have yet another class which enumerates higher level "builder" functions for every single "kind" of response (e.g. `buildSuccessfulResponse`, `buildFileNotFoundResponse`, `buildDeleteFailed(Exception)`, etc.). However, these builder methods would be only ever be used in this class, which feels to me like unnecessary indirection, and as a result, incidental complexity.

Reply:
However, I suppose signatures like 
```
Response performDeleteRequest(Request request)
``` 
both convey the intended side effect (deleting a file) and the result (an HTTP response to the client). I'll consider this for future refactors.